internal/provider/provider.go

```diff
@@ -355,9 +355,54 @@ func (p *SaladCloudProvider) createContainerGroup(createContainerList []saladcli
 	for _, container := range createContainerList {
 		createContainerGroupRequest := *saladclient.NewCreateContainerGroup(utils.GetPodName(pod.Namespace, pod.Name, pod), container, "always", 1)
 		createContainerGroupRequest.SetCountryCodes(countryCodesEnum)
+		readinessProbe, err := p.getWorkloadContainerProbeFrom(pod.Spec.Containers[0].ReadinessProbe)
+		if err == nil {
+			createContainerGroupRequest.ReadinessProbe = *readinessProbe
+		} else {
+			log.G(context.Background()).Errorf("Failed to get readinessProbe ", err)
+		}
+		livenessProbe, err := p.getWorkloadContainerProbeFrom(pod.Spec.Containers[0].LivenessProbe)
+		if err == nil {
+			createContainerGroupRequest.LivenessProbe = *livenessProbe
+		} else {
+			log.G(context.Background()).Errorf("Failed to get livenessProbe ", err)
+		}
+		startupProbe, err := p.getWorkloadContainerProbeFrom(pod.Spec.Containers[0].StartupProbe)
+		if err == nil {
+			createContainerGroupRequest.StartupProbe = *startupProbe
+		} else {
+			log.G(context.Background()).Errorf("Failed to get startupProbe ", err)
+		}
 		createContainerGroups = append(createContainerGroups, createContainerGroupRequest)
 	}
-
 	return createContainerGroups
 
 }
+
+func (p *SaladCloudProvider) getWorkloadContainerProbeFrom(k8sProbe *corev1.Probe) (*saladclient.NullableContainerGroupProbe, error) {
+	if k8sProbe == nil || *k8sProbe == (corev1.Probe{}) {
+		return nil, nil
+	}
+	if k8sProbe.GRPC != nil {
+		log.G(context.Background()).Errorf("GRPC Probe is not supported")
```

Comment:
There is support for gRPC probes in the API. Is there a reason to exclude it here?

Reply:
resolved